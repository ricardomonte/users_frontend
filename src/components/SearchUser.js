import React from 'react';
import TextField, {HelperText, Input} from '@material/react-text-field';
import '@material/react-text-field/dist/text-field.css';

const SearchUser = ({term, handleSearch}) => {
  return (
    <div style={{width: '100%'}}>
      <TextField
        label='Search User'
        helperText={<HelperText>ie. Max, Stainer, Samantha</HelperText>}
        style={{width: '100%'}}
      >
        <Input value={term} onChange={handleSearch} />
      </TextField>
    </div>
  )
}

export default SearchUser;