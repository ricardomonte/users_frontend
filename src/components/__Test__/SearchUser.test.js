import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import SearchUser from '../SearchUser';

test('Content from SearchUser component matches snapshot', () => {
  const content = render(
    <SearchUser handleSearch={jest.fn()} term={`david`} />
  );
  expect(content.container).toMatchSnapshot();
  expect(screen.getByDisplayValue('david')).toBeInTheDocument();
})