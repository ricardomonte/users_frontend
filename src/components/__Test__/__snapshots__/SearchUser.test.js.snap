// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content from SearchUser component matches snapshot 1`] = `
<div>
  <div
    class="w100 sticky"
  >
    <div
      class="mdc-text-field w100 mdc-text-field--with-trailing-icon"
    >
      <input
        class="mdc-text-field__input"
        id=""
        value="david"
      />
      <label
        class="mdc-floating-label mdc-floating-label--float-above"
        for=""
      >
        Search User
      </label>
      <div
        class="mdc-line-ripple"
      />
      <i
        class="material-icons mdc-text-field__icon"
        tabindex="-1"
      >
        search
      </i>
    </div>
    <div
      class="mdc-text-field-helper-line"
    >
      <p
        aria-hidden="false"
        class="mdc-text-field-helper-text"
      >
        ie. Max, Stainer, Samantha
      </p>
    </div>
  </div>
</div>
`;
